src/adstaylorinteg.jl

```diff
@@ -0,0 +1,327 @@
+    fit = curve_fit(exp_model!, exp_model_jacobian!, view(xs, idxs),
+                        view(ys, idxs), p0; inplace = true)
+    # Estimate next order coefficient
+    return exp_model(varorder+1, fit.param)
+end
+
+# Split node's domain in half
+# See section 3 of https://doi.org/10.1007/s10569-015-9618-3
+function split(node::ADSBinaryNode{N, M, T}, x::SVector{M, TaylorN{T}},
+               p::SVector{M, Taylor1{TaylorN{T}}}) where {N, M, T <: Real}
+    # Split direction
+    j = splitdirection(x)
+    # Split domain
+    s1, s2 = split(node.s, j)
+    # Jet transport variables
+    v_1, v_2 = get_variables(), get_variables()
+    # Shift expansion point
+    r = getroot(node).s
+    v_1[j] = v_1[j]/2 + r.lo[j]/2
+    v_2[j] = v_2[j]/2 + r.hi[j]/2
+    # Taylor1 order
+    order = p[1].order
+    # Left half
+    x1 = SVector{M, TaylorN{T}}(x[i](v_1) for i in eachindex(x))
+    p1 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_1), p[i].coeffs), order ) for i in eachindex(p)
+    )
+    # Right half
+    x2 = SVector{M, TaylorN{T}}(x[i](v_2) for i in eachindex(x))
+    p2 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_2), p[i].coeffs), order ) for i in eachindex(p)
+    )
+
+    return s1, x1, p1, s2, x2, p2
+end
+
+"""
+    split!(node::ADSBinaryNode{N, M, T}, p::SVector{M, Taylor1{TaylorN{T}}},
+           dt::T, nsplits::Int, maxsplits::Int, stol::T) where {N, M, T <: Real}
+
+Split `node` in half if any element of `p(dt)` has an `adsnorm` greater than `stol`.
+"""
+function split!(node::ADSBinaryNode{N, M, T}, p::SVector{M, Taylor1{TaylorN{T}}},
+                dt::T, nsplits::Int, maxsplits::Int, stol::T) where {N, M, T <: Real}
+    # Evaluate x at dt
+    x = _eval(p, dt)
+    # Split criteria for each element of x
+    mask = adsnorm.(x)
+    # Split
+    if nsplits < maxsplits && any(mask .> stol)
+        # Split
+        s1, x1, p1, s2, x2, p2 = split(node, x, p)
+        # Left half
+        leftchild!(node, s1, node.t + dt, x1, p1)
+        # Right half
+        rightchild!(node, s2, node.t + dt, x2, p2)
+        # Update number of splits
+        nsplits += 1
+    # No split
+    else
+        leftchild!(node, node.s, node.t + dt, x, p)
+    end
+
+    return nsplits
+end
+
+function taylorinteg(
+    f!, q0::SVector{M, TaylorN{T}}, s::ADSDomain{N, T}, t0::T, tmax::T,
+    order::Int, stol::T, abstol::T, params = nothing; maxsplits::Int = 10,
+    maxsteps::Int = 500, parse_eqs::Bool = true
+    ) where {N, M, T <: Real}
+
+    # Initialize the vector of Taylor1 expansions
+    t = t0 + Taylor1( T, order )
+    x = Array{Taylor1{TaylorN{T}}}(undef, M)
+    dx = Array{Taylor1{TaylorN{T}}}(undef, M)
+    @inbounds for i in eachindex(q0)
+        @inbounds x[i] = Taylor1( q0[i], order )
+        @inbounds dx[i] = Taylor1( zero(q0[i]), order )
+    end
+
+    # Determine if specialized jetcoeffs! method exists
+    parse_eqs, rv = _determine_parsing!(parse_eqs, f!, t, x, dx, params)
```

Comment:
Since the integration of the splits is done sequentially, we can reuse `rv`. In case we wanted to propagate the splits in parallel, we might need a `rv` per task.